src/test/java/org/opentripplanner/middleware/controllers/api/TrackedTripControllerTest.java

```diff
@@ -0,0 +1,226 @@
+        soloOtpUser = Persistence.otpUsers.getById(soloOtpUser.id);
+        if (soloOtpUser != null) soloOtpUser.delete(true);
+        monitoredTrip = Persistence.monitoredTrips.getById(monitoredTrip.id);
+        if (monitoredTrip != null) monitoredTrip.delete();
+    }
+
+    @AfterEach
+    public void tearDownAfterTest() {
+        if (trackedJourney != null) {
+            trackedJourney.delete();
+            trackedJourney = null;
+        }
+    }
+
+    @Test
+    void canCompleteJourneyLifeCycle() throws Exception {
+        assumeTrue(IS_END_TO_END);
+
+        var response = makeRequest(
+            START_TRACKING_TRIP_PATH,
+            JsonUtils.toJson(createInitialTrackingPayload()),
+            getMockHeaders(soloOtpUser),
+            HttpMethod.POST
+        );
+
+        var startTrackingResponse = JsonUtils.getPOJOFromJSON(response.responseBody, StartTrackingResponse.class);
+        trackedJourney = Persistence.trackedJourneys.getOneFiltered(eq("journeyId", startTrackingResponse.journeyId));
+        assertEquals(ManageTripTracking.TRIP_TRACKING_UPDATE_FREQUENCY_SECONDS, startTrackingResponse.frequencySeconds);
+        assertEquals(TripInstruction.GET_ON_BUS.name(), startTrackingResponse.instruction);
+        assertEquals(TripStatus.ON_TRACK.name(), startTrackingResponse.tripStatus);
+        assertEquals(HttpStatus.OK_200, response.status);
+
+        response = makeRequest(
+            UPDATE_TRACKING_TRIP_PATH,
+            JsonUtils.toJson(createUpdateTrackingPayload(startTrackingResponse.journeyId)),
+            getMockHeaders(soloOtpUser),
+            HttpMethod.POST
+        );
+
+        var updateTrackingResponse = JsonUtils.getPOJOFromJSON(response.responseBody, UpdateTrackingResponse.class);
+        assertEquals(TripInstruction.STAY_ON_BUS.name(), updateTrackingResponse.instruction);
+        assertEquals(TripStatus.ON_TRACK.name(), updateTrackingResponse.tripStatus);
+        assertEquals(HttpStatus.OK_200, response.status);
+
+        response = makeRequest(
+            END_TRACKING_TRIP_PATH,
+            JsonUtils.toJson(createEndTrackingPayload(startTrackingResponse.journeyId)),
+            getMockHeaders(soloOtpUser),
+            HttpMethod.POST
+        );
+        assertEquals(HttpStatus.OK_200, response.status);
+    }
+
+    @Test
+    void canNotUseUnassociatedTrip() throws Exception {
+        assumeTrue(IS_END_TO_END);
+
+        HttpResponseValues response = makeRequest(
+            START_TRACKING_TRIP_PATH,
+            JsonUtils.toJson(createInitialTrackingPayload("unassociated-trip-id")),
+            getMockHeaders(soloOtpUser),
+            HttpMethod.POST
+        );
+
+        var startTrackingResponse = JsonUtils.getPOJOFromJSON(response.responseBody, StartTrackingResponse.class);
+        assertEquals("Monitored trip is not associated with this user!", startTrackingResponse.message);
+        assertEquals(HttpStatus.BAD_REQUEST_400, response.status);
```

Comment:
Should the response be "forbidden" instead?

Reply:
Addressed in https://github.com/ibi-group/otp-middleware/pull/203/commits/ec0152118d75a99b1cdf92e774c3e088927e6273